<a href="https://colab.research.google.com/github/MiguelFigueroa02/MCPI_Student_Repo/blob/main/courses/1-Semester/fundamentals%20of%20statistics/Projects/Notebooks/Estad%C3%ADstica_Descriptiva_Visualizaci%C3%B3n.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<p align="center">
  <img src="https://raw.githubusercontent.com/MiguelFigueroa02/MCPI_Student_Repo/refs/heads/main/images/logo.png"
       width="30%">
</p>

#Fundamentos de Estadística
---
**Profesor:** Dr. Daniel de La Rosa Gómez\
**Estudiante:** Miguel Figueroa Aguilar


## Estadística Descriptiva
---



### Nombre y Fuente:
---
En este caso elegí el Dataset relacionado con la hipertensión Arterial en México a partir de la plataforma de [Kaggle](https://www.kaggle.com/datasets/frederickfelix/hipertensin-arterial-mxico)\
Este es un conglomerado a partir de información de salud de pacientes reunidos a partir de la información de la Encuesta Nacional de Salud y Nutrición, en donde se reune información de diferentes tipo de información biométrica que se irá explicando a detalle posteriormente para identificar riesgos de hipertensión.\

### ¿Qué representan los datos?
---


In [11]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import requests
import plotly.express as px

In [9]:
#Establecemos un formato adecuado de visualización de los dataframe que extraigamos
pd.set_option('display.max_columns', None)
pd.set_option('display.max_colwidth',None)
pd.set_option("display.float_format", lambda x: f"{x:,.2f}")

In [12]:
# El dataset se extrae de la plataforma de kaggle y se sube a un repositorio personal de github
url = 'https://raw.githubusercontent.com/MiguelFigueroa02/MCPI_Student_Repo/refs/heads/main/courses/1-Semester/fundamentals%20of%20statistics/Projects/Data/Hipertension_Arterial_Mexico.csv'

response = requests.get(url)
if response.status_code == 200:
  print('Se accedió a dataset con éxito')
  df = pd.read_csv(url)

Se accedió a dataset con éxito


In [14]:
df.head()

,FOLIO_I,sexo,edad,concentracion_hemoglobina,temperatura_ambiente,valor_acido_urico,valor_albumina,valor_colesterol_hdl,valor_colesterol_ldl,valor_colesterol_total,valor_creatina,resultado_glucosa,valor_insulina,valor_trigliceridos,resultado_glucosa_promedio,valor_hemoglobina_glucosilada,valor_ferritina,valor_folato,valor_homocisteina,valor_proteinac_reactiva,valor_transferrina,valor_vitamina_bdoce,valor_vitamina_d,peso,estatura,medida_cintura,segundamedicion_peso,segundamedicion_estatura,distancia_rodilla_talon,circunferencia_de_la_pantorrilla,segundamedicion_cintura,tension_arterial,sueno_horas,masa_corporal,actividad_total,riesgo_hipertension
0,2022_01001004,2,41,14.20,22,4.80,4.00,34,86.00,139,0.58,92.00,4.00,123,103,5.20,2.70,23.40,4.90,0.02,1.10,167.00,20.80,74.55,147.30,0.00,64.70,154.00,48.50,33.50,0.00,107,4,32.89,120,1
1,2022_01001009,2,65,14.10,9,4.40,3.80,73,130.00,252,0.62,111.00,9.40,131,131,6.20,2.70,23.40,4.90,0.02,1.10,167.00,20.80,2.00,151.00,0.00,96.75,152.20,44.50,41.10,113.70,104,2,1.00,240,0
2,2022_01001012,2,68,14.20,22,4.80,4.00,34,86.00,139,0.58,92.00,4.00,123,103,5.20,2.70,23.40,4.90,0.02,1.10,167.00,20.80,2.00,151.00,0.00,68.70,144.80,42.30,37.80,103.70,105,1,1.00,480,0
3,2022_01001013,1,35,15.70,11,6.50,4.10,49,107.00,203,0.91,106.00,7.40,297,105,5.30,2.70,23.40,4.90,0.02,1.10,167.00,20.80,81.50,173.40,98.50,64.70,154.00,48.50,33.50,0.00,117,5,26.27,275,1
4,2022_01001015,2,65,12.70,7,4.20,4.20,41,76.00,145,0.71,109.00,16.90,136,154,7.00,2.70,23.40,4.90,0.02,1.10,167.00,20.80,2.00,151.00,0.00,97.15,161.30,49.60,42.00,118.90,123,2,1.00,255,0


In [74]:
column_names = [f'`{column}`' for column in df.columns]
column_types = [f'`{df[column].dtype}`' for column in df.columns]
column_shapes = [df['sexo'].shape[0] for column in df.columns]
column_na = [df[column].isna().sum() for column in df.columns]

In [67]:
column_categories = []
for column in df.columns:
  if ((column == 'sexo') or (column == 'riesgo_hipertension')):
    column_categories.append('categórica nominal')
  else:
    if df[column].dtype == 'float64':
      column_categories.append('numérica continua')
    elif df[column].dtype == 'int64':
      column_categories.append('numérica discreta')
    else:
      column_categories.append('N/A')

In [76]:
pd.DataFrame({
    'Columna': column_names,
    'Tipos':column_types,
    'Num_Registros': column_shapes,
    'Tipo_Variable':column_categories
    }).to_markdown()

'|    | Columna                            | Tipos     |   Num_Registros | Tipo_Variable      |\n|---:|:-----------------------------------|:----------|----------------:|:-------------------|\n|  0 | `FOLIO_I`                          | `object`  |            4363 | N/A                |\n|  1 | `sexo`                             | `int64`   |            4363 | categórica nominal |\n|  2 | `edad`                             | `int64`   |            4363 | numérica discreta  |\n|  3 | `concentracion_hemoglobina`        | `float64` |            4363 | numérica continua  |\n|  4 | `temperatura_ambiente`             | `int64`   |            4363 | numérica discreta  |\n|  5 | `valor_acido_urico`                | `float64` |            4363 | numérica continua  |\n|  6 | `valor_albumina`                   | `float64` |            4363 | numérica continua  |\n|  7 | `valor_colesterol_hdl`             | `int64`   |            4363 | numérica discreta  |\n|  8 | `valor_colesterol_ldl`          

|    | Columna                            | Tipos     |   Num_Registros | Tipo_Variable      | Función   |
|---:|:-----------------------------------|:----------|----------------:|:-------------------| :-------------------|
|  0 | `FOLIO_I`                          | `object`  |            4363 | N/A                | |
|  1 | `sexo`                             | `int64`   |            4363 | categórica nominal |
|  2 | `edad`                             | `int64`   |            4363 | numérica discreta  |
|  3 | `concentracion_hemoglobina`        | `float64` |            4363 | numérica continua  |
|  4 | `temperatura_ambiente`             | `int64`   |            4363 | numérica discreta  |
|  5 | `valor_acido_urico`                | `float64` |            4363 | numérica continua  |
|  6 | `valor_albumina`                   | `float64` |            4363 | numérica continua  |
|  7 | `valor_colesterol_hdl`             | `int64`   |            4363 | numérica discreta  |
|  8 | `valor_colesterol_ldl`             | `float64` |            4363 | numérica continua  |
|  9 | `valor_colesterol_total`           | `int64`   |            4363 | numérica discreta  |
| 10 | `valor_creatina`                   | `float64` |            4363 | numérica continua  |
| 11 | `resultado_glucosa`                | `float64` |            4363 | numérica continua  |
| 12 | `valor_insulina`                   | `float64` |            4363 | numérica continua  |
| 13 | `valor_trigliceridos`              | `int64`   |            4363 | numérica discreta  |
| 14 | `resultado_glucosa_promedio`       | `int64`   |            4363 | numérica discreta  |
| 15 | `valor_hemoglobina_glucosilada`    | `float64` |            4363 | numérica continua  |
| 16 | `valor_ferritina`                  | `float64` |            4363 | numérica continua  |
| 17 | `valor_folato`                     | `float64` |            4363 | numérica continua  |
| 18 | `valor_homocisteina`               | `float64` |            4363 | numérica continua  |
| 19 | `valor_proteinac_reactiva`         | `float64` |            4363 | numérica continua  |
| 20 | `valor_transferrina`               | `float64` |            4363 | numérica continua  |
| 21 | `valor_vitamina_bdoce`             | `float64` |            4363 | numérica continua  |
| 22 | `valor_vitamina_d`                 | `float64` |            4363 | numérica continua  |
| 23 | `peso`                             | `float64` |            4363 | numérica continua  |
| 24 | `estatura`                         | `float64` |            4363 | numérica continua  |
| 25 | `medida_cintura`                   | `float64` |            4363 | numérica continua  |
| 26 | `segundamedicion_peso`             | `float64` |            4363 | numérica continua  |
| 27 | `segundamedicion_estatura`         | `float64` |            4363 | numérica continua  |
| 28 | `distancia_rodilla_talon`          | `float64` |            4363 | numérica continua  |
| 29 | `circunferencia_de_la_pantorrilla` | `float64` |            4363 | numérica continua  |
| 30 | `segundamedicion_cintura`          | `float64` |            4363 | numérica continua  |
| 31 | `tension_arterial`                 | `int64`   |            4363 | numérica discreta  |
| 32 | `sueno_horas`                      | `int64`   |            4363 | numérica discreta  |
| 33 | `masa_corporal`                    | `float64` |            4363 | numérica continua  |
| 34 | `actividad_total`                  | `int64`   |            4363 | numérica discreta  |
| 35 | `riesgo_hipertension`              | `int64`   |            4363 | categórica nominal |